tools/codegen/dest/register_dispatch_key.py

```diff
@@ -339,20 +339,25 @@ def generate_defn(cpp_sig: CppSignature) -> str:
                 args_exprs_str = ', '.join(a.name for a in args)
 
                 device_check = '  // No device check\n'
-                if is_cuda_dispatch_key(self.backend_index.dispatch_key):
+                # Backends that require device guards presumably also require device checks.
+                if self.backend_index.device_guard:
                     device_check_args = itertools.chain(
                         f.func.arguments.out,
                         f.func.arguments.flat_positional
                     )
                     device_check = RegisterDispatchKey.gen_device_check(f.device_check, list(device_check_args), name)
 
                 device_guard = "// DeviceGuard omitted"  # default
-                if f.device_guard and is_cuda_dispatch_key(self.backend_index.dispatch_key):
+                if f.device_guard and self.backend_index.device_guard:
                     has_tensor_options = any(isinstance(a.argument, TensorOptionsArguments) for a in args)
                     if has_tensor_options:
                         # kernel is creating a tensor
-                        device_guard = """globalContext().lazyInitCUDA();
+                        device_guard = """
   const DeviceGuard device_guard(device_or_default(device));"""
+
+                        # CUDA requires special handling
```

Comment:
Other backend should be able to optin to this lazy init as well? But that can be done later.

Reply:
yep that's right - it seems like in the long term we'd want some sort of unified API for backends to perform lazy init. Intel doesn't currently need one, so we can push that off for later.